src/Transport/TestTransport.php

```diff
@@ -228,7 +233,23 @@ public function get(): iterable
             return [];
         }
 
-        return [\array_shift(self::$queue[$this->name])];
+        if (!$this->clock) {
+            return \array_shift(self::$queue[$this->name]);
+        }
+
+        $now = $this->clock->now();
+
+        foreach (self::$queue[$this->name] as $i => $envelope) {
+            if (($availableAtStamp = $envelope->last(AvailableAtStamp::class)) && $now > $availableAtStamp->getAvailableAt()) {
+                continue;
+            }
+
+            unset(self::$queue[$this->name][$i]);
+
+            return [$envelope];
+        }
```

Comment:
not sure if it is the good approach, but if a "no delayed" message follows a "delayed" message which is still not available, it would be processed first.

Reply:
totally, but this changes the behavior of this library

Reply:
Another reason to opt in